In [4]:
#install.packages("kableExtra")
#install.packages("tidyverse")

library(tidyverse)
library(knitr)
library(kableExtra)
library(IRdisplay)
library(dplyr)
library(gnm)
library(dplyr)

In [2]:
data <- read.csv2("Ankieta.csv")
data <- mutate(data, across(c("SEN", "BIEGANIE", "PIES")))
data <- ftable(data)
data.df <- as.data.frame(data)

data.df

SEN BIEGANIE PIES Freq
1 0   0        0     6  
2 1   0        0     5  
3 0   1        0     1  
4 1   1        0     5  
5 0   0        1     2  
6 1   0        1     5  
7 0   1        1     2  
8 1   1        1    14

___
___
___

## Zadanie 3.

W zadaniu trzecim testujemy hipotezy dotyczące parametrów modelu log-liniowego hierarchicznie uporządkowanego na podstawie tych samych danych, co poprzednio.

### a.
W podpunkcie a przeprowadzamy test, w którym hipoteza zerowa $H_0$ mówi, że zmienne Sen, Bieganie i Pies są niezależne (tj. pochodzą z modelu $[1\,2\,3]$). Z kolei, hipoteza alternatywna $H_a$ mówi, że zmienne pochodzą z modelu pełnego. Poziom istotności ustalamy jako $\alpha = 0.05$.

In [53]:
model_a <- glm(Freq ~ SEN + BIEGANIE + PIES, 
            data = data.df, family = poisson)

In [24]:
p_a <- 1-pchisq(deviance(model_a), df = df.residual(model_a))

p_a %>% round(4)

[1] 0.0293

P-wartość w tym teście jest mniejsza od $\alpha$, więc odrzucamy $H_0$.

#### Wersja z nadmodelem
Przeprowadźmy dodatkowo wersję tego testu, gdzie zmienimy hipotezę alternatywną. Niech $H_a$ oznacza, że dane pochodzą z nadmodelu $[12\,13\,23]$.

In [37]:
model_a_2 <- glm(Freq ~ (SEN + BIEGANIE + PIES)^2, 
            data = data.df, family = poisson)

test <- anova(model_a, model_a_2)
p_a_2 <- 1-pchisq(test$Deviance[2], df = test$Df[2])

p_a_2 %>% round(4)

[1] 0.0144

Otrzymujemy p-wartość na poziomie $0.0144$, więc ponownie jest mniejsza od $\alpha$. Zatem niezależnie od hipotezy alternatywnej $H_a$, możemy stwierdzić, że dane nie pochodzą z modelu $[1\,2\,3]$.

### b.
W podpunkcie b, hipoteza zerowa $H_0$ mówi nam, że zmienna Pies jest niezależna od niezależna od pary zmiennych Sen i Bieganie. Jest to równoważne stwierdzeniu, że dane pochodzą z modelu $[12\,\,3].$ Najpierw wykonajmy wersję z hipotezą alternatywną $H_a: \text{dane pochodzą z modelu pełnego.}$

In [54]:
model_b <- glm(Freq ~ PIES + SEN*BIEGANIE, 
            data = data.df, family = poisson)

In [55]:
p_b <- 1-pchisq(deviance(model_b), df = df.residual(model_b))

p_b %>% round(4)

[1] 0.1132

Otrzymana p-wartość wynosi $0.1132 > 0.05$, więc nie mamy podstaw do odrzucenia $H_0$.

#### Wersja z nadmodelem
Ponownie, niech $H_a$ oznacza, że dane pochodzą z nadmodelu $[12\,13\,23]$.

In [52]:
model_b_2 <- glm(Freq ~ (SEN + BIEGANIE + PIES)^2, 
            data = data.df, family = poisson)

test <- anova(model_b, model_b_2)
p_b_2 <- 1-pchisq(test$Deviance[2], df = test$Df[2])

p_b_2 %>% round(4)

[1] 0.0562

Widzimy, że niezależnie od doboru $H_a$, nie mamy podstaw do odrzucenia $H_0$.

### c.
Hipoteza zerowa $H_0$ z podpunktu c stanowi, że zmienna Sen jest niezależna od zmiennej Pies przy ustalonej zmiennej Bieganie. Jest to równoważne ze stwierdzeniem, że dane pochodzą z modelu $[12\,\,23].$ Najpierw przeprowadźmy test z hipotezą alternatywną $H_a$, że dane pochodzą z modelu pełnego.

In [48]:
model_c <- glm(Freq ~ SEN*BIEGANIE + PIES*BIEGANIE,
            data = data.df, family = poisson)

In [49]:
p_c <- 1-pchisq(deviance(model_c), df = df.residual(model_c))

p_c %>% round(4)

[1] 0.5329

P-wartość wynosi $0.5329 > 0.05$, więc ponownie nie mamy przesłanek do odrzucenia $H_0$.

#### Wersja z nadmodelem
Zobaczmy jeszcze wersję z hipotezą alternatywną $H_a$, że dane pochodzą z nadmodelu $[12\,13\,23]$.

In [43]:
model_c_2 <- glm(Freq ~ (SEN + BIEGANIE + PIES)^2, 
            data = data.df, family = poisson)

test <- anova(model_c, model_c_2)
p_c_2 <- 1-pchisq(test$Deviance[2], df = test$Df[2])

p_c_2 %>% round(4)

[1] 0.3058

Ponownie widzimy, że niezależnie od doboru $H_a$, dochodzimy do takich samych wniosków.
